.github/workflows/style_check.yml

```diff
@@ -0,0 +1,47 @@
+name: Style check
+
+on:
+  # Trigger on pull requests.
+  pull_request:
+
+  # Trigger on pushes to the mainline branches and version tags. This prevents building commits
+  # twice when the pull request source branch is in the same repository.
+  push:
+    branches:
+      - "maint"
+      - "master"
+
+  # Trigger on request.
+  workflow_dispatch:
+
+jobs:
+  style_checks:
+    name: Run ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    container:
+      image: glotzerlab/ci:2021.03-clang10_py38
+      options: -u 0
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: clang_format
+            command: ./run-clang-format.py -r .
+          - name: flake8
+            command: flake8
+          - name: yapf
+            command: yapf -d -r .
```

Comment:
If we decide against using pre-commit, I would advocate for combining the Python style checks. Run yapf first, then flake8. That way you ensure that the result of the formatting _pipeline_ is good, instead of having one action (yapf) reformat files and another action (flake8) validate them.

Reply:
This invocation of `yapf` does not modify any files (the `-i` option is missing). It only prints the diff of what it would modify.